src/vtquery.cpp

```diff
@@ -367,163 +404,61 @@ struct Worker : Nan::AsyncWorker {
+                                if (value_is_duplicate(result, feature, layer_name, original_geometry_type, comparison_tags)) {
+                                    if (meters <= result.distance) {
+                                        auto props = mapbox::vector_tile::extract_properties(feature);
+                                        insert_result(result, props, comparison_tags, layer_name, ll, meters, original_geometry_type, feature.has_id(), feature.id());
+                                        found_duplicate = true;
+                                        break;
+                                    // if we have a duplicate but it's lesser than what we already have, just skip and don't add below
+                                    // we need to set skip_duplicate to true because found_duplicate will still be false
+                                    } else {
+                                        skip_duplicate = true;
+                                        break;
                                     }
                                 }
                             }
+                        }
 
-                            // do not add to results
-                            if (duplicate_skip) {
-                              continue;
-                            }
+                        if (skip_duplicate) {
+                            continue;
+                        }
 
-                            // remove references that were marked as lesser duplicates
-                            if (references_for_removal.size() > 0) {
-                                for (auto i : references_for_removal) {
-                                    results_queue_.erase(i);
-                                }
-                            }
+                        if (found_duplicate) {
+                            std::sort(results_queue_.begin(), results_queue_.end(), CompareDistance());
+                            continue;
+                        }
 
-                            results_.emplace_back(std::move(props),
-                                                  std::move(comparison_tags),
-                                                  layer_name,
-                                                  std::move(qp),
-                                                  0.0,
-                                                  original_geometry_type,
-                                                  feature.has_id(),
-                                                  feature.id());
-                            results_queue_.emplace_back(&results_.back());
+                        if (meters < results_queue_.back().distance) {
+                            auto props = mapbox::vector_tile::extract_properties(feature);
+                            insert_result(results_queue_.back(), props, comparison_tags, layer_name, ll, meters, original_geometry_type, feature.has_id(), feature.id());
```

Comment:
@flippmoke should I be able to `std::move(props)` and `std::move(comparison_tags)` here?

Reply:
error is `candidate function not viable: expects an l-value for 2nd argument`